source/mir/random/discrete.d

```diff
@@ -88,36 +78,37 @@ struct Discrete(T)
     size_t opCall(RNG)(ref RNG gen) const
     {
         import std.random : uniform;
-        T v = uniform!("[)", T, T)(0, _cdPoints[$-1], gen);
-        return (cast(SortedRange!(const(T)[], "a <= b")) r).lowerBound(v).length;
+        import std.range : assumeSorted;
+
+        T v = uniform!("[)", T, T)(0, cdPoints[$-1], gen);
```

Comment:
Am I right in thinking that `a < b` doesn't allow for any discrete element to have zero probability, whereas `a <= b` does ... ?